src/mount/mount-tool.c

```diff
@@ -914,6 +910,28 @@ static int action_umount(
 
         int i, r, r2 = 0;
 
+        if (arg_transport != BUS_TRANSPORT_LOCAL) {
+                for (i = optind; i < argc; i++) {
+                        _cleanup_free_ char *p = NULL;
+
+                        p = strdup(argv[i]);
+                        if (!p)
+                                return log_oom();
+
+                        path_kill_slashes(p);
+
+                        if (!path_is_absolute(p)) {
+                                log_error("Only absolute path is supported: %s", p);
+                                r2 = -EINVAL;
```

Comment:
You continue with unmounting the path even if the check fails and error is emitted.

Maybe it'd be better to even move all the checks up front, and refuse all operations if at least one of the arguments is wrong? It's pretty annoying to authenticate yourself to a remote server only to have the operation fail halfway.